tests/lib/prepare.sh

```diff
@@ -14,29 +14,34 @@ update_core_snap_with_snap_exec_snapctl() {
     snap="$(mount | grep " $core" | awk '{print $1}')"
     umount --verbose "$core"
 
-    # Now unpack the core, inject the new snap-exec and snapctl into it, and
-    # repack it.
+    # Now unpack the core, inject the new snap-exec/snapctl into it
     unsquashfs "$snap"
     cp /usr/lib/snapd/snap-exec squashfs-root/usr/lib/snapd/
     cp /usr/bin/snapctl squashfs-root/usr/bin/
+    # also add snap/snapd because we re-exec by default.
```

Comment:
I'll say this again just for completeness. I think that ultimately reexec should be fully symmetric.

If it is enabled we always take snapd/snap/snap-confine/snapctl/snap-exec from the core snap, if it is more recent than from the pacakge. If it is disabled we always take snapd/snap-confine/snapctl/snap-exec from the classic package. Why? Because this ensures that the set is matched. This especially matters for snapctl and snap-exec as we learned in the past.

I think that re-packaging aside. We should use snap-confine to ensure this policy is used.